docs/DeveloperGuide.md

```diff
@@ -896,41 +893,50 @@ Use case ends.
 
 **MSS**
 
-1. User requests to list unassigned residents and unallocated rooms.
-2. SunRez displays unassigned residents and unallocated rooms side-by-side.
-3. User requests to allocate a room to a resident.
-4. SunRez allocates the room and saves the changes.
+1. User requests to list unassigned residents.
+2. SunRez displays unassigned residents.
+3. User requests to allocate a resident to a room.
+4. SunRez allocates the resident to the room and saves the changes.
 
 **Extensions**
 
-* 2a. There are no unassigned residents or unassigned.
+* 2a. There are no unassigned residents.
     
     Use case ends.
-* 3a. The given index is invalid.
-    * 3a1. SunRez shows an error message.
+* 3a. The given resident or room does not exist.
+    * 3a1. SunRez shows an error message.   
+      
+        Use case resumes at step 2.
 
+* 3b. The given resident or room have already been assigned.
+    * 3b1. SunRez shows an error message. 
+      
         Use case resumes at step 2.
 
 
 ### UC-020 Deallocate a Room
 
 **MSS**
 
-1. User requests to list room allocations.
-2. SunRez displays rooms, each with its corresponding allocated resident.
-3. User requests to deallocate a room based on the index from the list displayed in step 2.
-4. SunRez deallocates the room and saves the changes.
+1. User requests to list the residents.
+2. SunRez displays residents, each with its corresponding allocated room, if any.
+3. User requests to deallocate a resident from a room based on the index from the list displayed in step 2.
```

Comment:
Hmm, this seems like it has too much detail about the implementation.

Suggestion: remove the stuff about index and list displayed. Keep it high level like UC-019.

Reply:
Good point. Thanks.